jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/FileBufferedResponseHandler.java

```diff
@@ -193,41 +215,154 @@ private void commit(Callback callback)
             }
 
             // Create an iterating callback to do the writing
-            IteratingCallback icb = new IteratingCallback()
+            try
+            {
+                SendFileCallback sfcb = new SendFileCallback(this, _filePath, getBufferSize(), callback);
+                sfcb.setUseFileMapping(isUseFileMapping());
+                sfcb.iterate();
+            }
+            catch (IOException e)
             {
-                private final long fileLength = _filePath.toFile().length();
-                private long _pos = 0;
-                private boolean _last = false;
+                callback.failed(e);
+            }
+        }
+    }
+
+    // TODO: can this be made generic enough to put into jetty-io somewhere?
```

Comment:
To answer your comment: I don't think so. We do 99% of our IO's using `Content.Source` and `Content.Sink` for which we already have `Content.copy()`.

Reply:
Implemented `Content.copy()` technique instead.